File: src/main/scala/com/github/mauricio/postgresql/ExecutorServiceUtils.scala
package com.github.mauricio.postgresql

import java.util.concurrent.Executors
import util.DaemonThreadsFactory

/**
 * User: Maurício Linhares
 * Date: 3/5/12
 * Time: 9:25 PM
 */

object ExecutorServiceUtils {
  val CachedThreadPool = Executors.newCachedThreadPool( DaemonThreadsFactory )
}

File: db-async-common/src/main/java/com/github/jasync/sql/db/util/ExecutorServiceUtils.kt
package com.github.jasync.sql.db.util

import java.util.concurrent.ExecutorService
import java.util.concurrent.Executors
import java.util.concurrent.ForkJoinPool

@Suppress("unused")
object ExecutorServiceUtils {

    val CommonPool: ExecutorService = ForkJoinPool.commonPool()

    fun newFixedPool(count: Int, name: String): ExecutorService {
        return Executors.newFixedThreadPool(count, DaemonThreadsFactory(name))
    }

}
